.buildkite/ml.rayci.yml

```diff
@@ -40,6 +52,29 @@ steps:
         --except-tags gpu_only,gpu,minimal,tune,doctest,needs_credentials 
     depends_on: [ "mlbuild", "forge" ]
 
+  - label: ":train: ml: {{matrix.python}} tests ({{matrix.worker_id}})"
+    key: ml_python_tests
+    if: build.pull_request.labels includes "continuous-build" || pipeline.id == "0189e759-8c96-4302-b6b5-b4274406bf89" || pipeline.id == "018f4f1e-1b73-4906-9802-92422e3badaa"
+    tags: 
+      - python
+      - train
+      - tune
+      - ml
+    instance_type: large
+    commands:
+      - bazel run //ci/ray_ci:test_in_docker -- //python/ray/train/... //python/ray/tune/... //python/ray/air/... ml
+        --workers 4 --worker-id {{matrix.worker_id}} --parallelism-per-worker 3
+        --python-version {{matrix.python}}
+        --except-tags gpu_only,gpu,minimal,doctest,needs_credentials,soft_imports,rllib,multinode
+    depends_on: 
+      - mlbuild-multipy
+      - forge
+    job_env: 
+    matrix:
+      setup:
+        python: ["3.12"]
+        worker_id: ["0", "1", "2", "3"]
```

Comment:
why we need this kind of matrix here?

Reply:
it's the number of build-kite jobs required, computed by the number of python versions x number of test shards required for each python

```
shard/python   3.10  3.11  3.12
0               X.      X.     X
1                          ...
2
3
```